admin/ajax/class-yoast-plugin-conflict-ajax.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @package WPSEO\admin|ajax
+ */
+
+/**
+ * Class Yoast_Plugin_Conflict_Ajax
+ */
+class Yoast_Plugin_Conflict_Ajax {
+
+	/**
+	 * @var string
+	 */
+	private $option_name = 'wpseo_dismissed_conflicts';
+
+	/**
+	 * @var array
+	 */
+	private $dismissed_conflicts = array();
+
+	/**
+	 * Initialize the hooks for the AJAX request
+	 */
+	public function __construct() {
+		add_action( 'wp_ajax_wpseo_dismiss_plugin_conflict', array( $this, 'dismiss_notice' ) );
+	}
+
+	/**
+	 * Handles the dismiss notice request
+	 */
+	public function dismiss_notice() {
+		check_ajax_referer( 'dismiss-plugin-conflict' );
+
+		$conflict_data = filter_input( INPUT_POST, 'data', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
+
+		$this->dismissed_conflicts = $this->get_dismissed_conflicts( $conflict_data['section'] );
+
+		$this->compare_plugins( $conflict_data['plugins'] );
+
+		$this->save_dismissed_conflicts( $conflict_data['section'] );
+
+		wp_die( 'true' );
+	}
+
+	/**
+	 * Getting the user option from the database
+	 *
+	 * @return bool|array
+	 */
+	private function get_dismissed_option() {
+		return get_user_option( $this->option_name, get_current_user_id() );
+	}
+
+	/**
+	 * Getting the dismissed conflicts from the database
+	 *
+	 * @param string $plugin_section
+	 *
+	 * @return array
+	 */
+	private function get_dismissed_conflicts( $plugin_section ) {
+		$dismissed_conflicts = $this->get_dismissed_option();
+
+		if ( is_array( $dismissed_conflicts ) &&  array_key_exists( $plugin_section, $dismissed_conflicts ) ) {
```

Comment:
I'd say the default for this should be empty array. Then you don't have to do the is_array check..